### DISCUSSION TITLE
enable strict peer deps for monorepo packages

### DISCUSSION THREAD

**Discussion by qhantom** [pnpm/pnpm#6029] [Q&A]
Hey there!

We've got a monorepo with several packages. One of the package (say it's called A) contains peer deps that are written in the `package.json` file. Another package (say it's called B) does import package A as (prod) dep. In order to make package A work, we need to install all peer deps. I recognized that I don't get any warning when I `pnpm install` all deps of the monorepo if I install the wrong version of the peer deps that package A requires. That led to bugs in production but I couldn't catch them in advance. The package A ist imported as `"lib": "workspace:*"`.  The strict peer deps flag does not have any effect.

Thanks for your help!

**Comment by zkochan** [comment]
Try to use [injected](https://pnpm.io/package_json#dependenciesmetainjected) in package B. So add this to the package.json of B:

```
{
  "dependenciesMeta": {
    "A": {
      "injected": true
    }
  }
}
```

**Reply by qhantom** [reply]
Thanks @zkochan! That helped but I realized that if I built my app, the package A (called `lib` here) isn't found:

![image](https://user-images.githubusercontent.com/54100938/216556142-74cf9420-af48-42cd-be39-88d53afd64bd.png)

This is what I added:

```
  "dependenciesMeta": {
    "lib": {
      "injected": true
    }
  }
```
